test/reduce.jl

```diff
@@ -94,6 +94,13 @@ end
 
 @test sum_kbn([1,1e100,1,-1e100]) == 2
 @test sum_kbn(Float64[]) == 0.0
+@test sum_kbn(i for i=1.0:1.0:10.0) == 55.0
```

Comment:
it would be worth making these use `===` instead of `==` so they also check that the output type is as expected